include/mmtf/binary_decoder.hpp

```diff
@@ -172,23 +203,22 @@ inline BinaryDecoder::BinaryDecoder(const msgpack::object& obj,
         err << "The '" + key + "' entry is too short " << obj.via.bin.size;
         throw DecodeError(err.str());
     }
-    // get data (encoded data is only pointed to and not parsed here)
-    const char* bytes = obj.via.bin.ptr;
+    this->initFromData(obj.via.bin.ptr, obj.via.bin.size);
+}
 
-    assignBigendian4(&strategy_, bytes);
-    assignBigendian4(&length_, bytes + 4);
-    assignBigendian4(&parameter_, bytes + 8);
-    encodedData_ = bytes + 12;
-    encodedDataLength_ = obj.via.bin.size - 12;
+inline BinaryDecoder::BinaryDecoder(const std::string& str,
+                                    const std::string& key)
+                                   : key_(key) {
+    this->initFromData(str.data(), str.size());
```

Comment:
This looks a bit dangerous. The documentation should emphasize that `str` must outlive the `BinaryDecoder` instance, because the `BinaryDecoder` directly views the string's data.